core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -98,35 +151,73 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
      * @param lon The longitude
      */
     public GeoBoundingBoxQueryBuilder bottomLeft(double lat, double lon) {
-        box[BOTTOM] = lat;
-        box[LEFT] = lon;
+        if (topLeft != null) {
+            topLeft.resetLon(lon);
+        } else {
+            topLeft = new GeoPoint(Double.NaN, lon);
+        }
+
+        if (bottomRight != null) {
+            bottomRight.resetLat(lat);
+        } else {
+            bottomRight = new GeoPoint(lat, Double.NaN);
+        }
```

Comment:
No it's because we are supporting both - specifying a bounding box with topRight/bottomLeft corners as well as topLeft/bottomRight corners.

I think I can make it slightly easier by always initialising the respective points. What will remain is having to map from top/left/right/bottom to latitude/longitude which is sometimes confusing for me.